src/cmdhandler/command_handler.cpp

```diff
@@ -203,29 +205,19 @@ std::string command_handler::state_system(const std::vector<std::string>& tokens
             current_state.collectors.clear();
             return "current state has been cleared";
         }else{
-            if(remove(("states/"+state_to_delete).c_str()) != 0){
-                return "saved state '"+state_to_delete+"' does not exist";
-            }else{
+            if(std::filesystem::remove((SAVE_STATE_DIR+state_to_delete).c_str())){
                 return "state '"+state_to_delete+"' has been removed";
+            }else{
+                return "saved state '"+state_to_delete+"' does not exist";
             }
         }
     }else if(tokens[0] == "list"){
         std::string response = "Saved states:";
 
-        DIR *states_dir = opendir("states/");
-        struct dirent *dir_read;
-
         //get file names in the 'states' directory
-        if(states_dir != nullptr){
-            while((dir_read = readdir(states_dir)) != nullptr){
-                //remove the .. and . identifiers
-                if(std::strcmp(dir_read->d_name, "..") != 0 && std::strcmp(dir_read->d_name, ".") != 0){
-                    response += "\n    "+std::string(dir_read->d_name);
-                }
-            }
-            closedir(states_dir);
-        }else{
-            return "The 'states' directory doesn't exist";
+        for (const auto & entry : std::filesystem::directory_iterator(SAVE_STATE_DIR)){
+            std::string filename_string = std::string(entry.path());
```

Comment:
Use `entry.path().string()` instead of `std::string(entry.path())` for getting a path as a string